prime_factor/prime_factor.c

```diff
@@ -0,0 +1,37 @@
+#include <stdio.h>
+
+void generate_primes(int *ar, int *cnt) {
```

Comment:
The function should return an `int`. It feels absurd that a function is void and it needs to *give back* an integer and it is using a pointer based parameter for that.